pre_commit_hooks/jira_pre_commit/main.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+import argparse
+import io
+import re
+import sys
+
+# Regex pattern to match a JIRA Task ID (e.g. SD-373)
+GIT_COMMIT_REGEX = re.compile(r"(?<![A-Z])[A-Z]{2,11}(?![A-Z])-(?<![0-9])[0-9]{1,6}(?![0-9])")
+# Error message printed when no JIRA Task ID is found
+ERROR_MSG = "[ERROR] Aborting commit. Your commit message is missing a Jira Task ID, i.e. JIRA-1234."
+
+
+def parse_args(argv=None):
+    """Provide CLI for jira_pre_commit hook."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "filename",
+        nargs=1,
+        help="Path to `COMMIT_EDITMSG` file.",
+    )
+    parser.add_argument(
+        "--exclude-pattern", "-e",
+        action="append",
+        default=['^Merge '],
+        help=(
+            "Regex to exclude commit messages from Jira Task ID check. "
+            "Can be specified multiple times. Default is '^Merge '."
+        ),
+    )
+
+    return parser.parse_args(argv)
+
+
+def check_exclusions(commit_message: str, patterns: list) -> int:
```

Comment:
1. Let's rename
2. patterns - should be list of strings, pls specify it in param annotation everywhere
```suggestion
def is_commit_excluded(commit_message: str, patterns: list[str]) -> int:
```

Reply:
Done